File: library/src/main/java/com/smartmobilefactory/epubreader/utils/BaseDisposableObserver.java
package com.smartmobilefactory.epubreader.utils;

import io.reactivex.SingleObserver;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.observers.DisposableObserver;

public class BaseDisposableObserver<E> extends DisposableObserver<E> implements SingleObserver<E> {
    @Override
    public void onNext(E o) {

    }

    @Override
    public void onSuccess(E e) {

    }

    @Override
    public void onError(Throwable e) {

    }

    @Override
    public void onComplete() {

    }

    public void addTo(CompositeDisposable compositeDisposable) {
        compositeDisposable.add(this);
    }
}

File: library/src/main/java/com/smartmobilefactory/epubreader/utils/BaseDisposableObserver.kt
package com.smartmobilefactory.epubreader.utils

import io.reactivex.SingleObserver
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.observers.DisposableObserver

internal class BaseDisposableObserver<E> : DisposableObserver<E>(), SingleObserver<E> {
    override fun onNext(o: E) {

    }

    override fun onSuccess(e: E) {

    }

    override fun onError(e: Throwable) {
        e.printStackTrace()
    }

    override fun onComplete() {

    }

    fun addTo(compositeDisposable: CompositeDisposable) {
        compositeDisposable.add(this)
    }
}
